sw/host/provisioning/orchestrator/src/ot_dut.py

```diff
@@ -115,11 +114,37 @@ def run_cp(self) -> None:
         # Run provisioning flow and collect logs.
         res = run(cmd, f"{self.log_dir}/cp_out.log.txt",
                   f"{self.log_dir}/cp_err.log.txt")
+
         if res.returncode != 0:
             logging.warning(f"CP failed with exit code: {res.returncode}.")
             confirm()
+
+        with open(f"{self.log_dir}/cp_out.log.txt", "r") as f:
+            log_data = f.read()
+
+        pattern = r'CHIP_PROBE_DATA:\s*({.*?})'
+        match = re.search(pattern, log_data)
+        if match:
+            json_string = match.group(1)
+            try:
+                chip_probe_data = json.loads(json_string)
+                logging.info(f"CHIP_PROBE_DATA: {chip_probe_data}")
+            except json.JSONDecodeError as e:
+                logging.error(f"Failed to parse CHIP_PROBE_DATA: {e}")
+                confirm()
         else:
-            logging.info("CP completed successfully.")
+            logging.error("CHIP_PROBE_DATA not found.")
+            confirm()
+
+        if "cp_device_id" not in chip_probe_data:
+            logging.error("cp_device_id found in CHIP_PROBE_DATA.")
+            confirm()
+
+        logging.info(
+            f"Updating device ID to: {chip_probe_data['cp_device_id']}")
+        self.device_id.from_hexstr(chip_probe_data["cp_device_id"])
```

Comment:
Note: this will just create the first 128-bits of the device ID (the "_base_uid" [portion](https://cs.opensource.google/opentitan/opentitan/+/earlgrey_1.0.0:sw/host/provisioning/orchestrator/src/device_id.py;l=121)); this needs to be combined with the SKU specific [part](https://cs.opensource.google/opentitan/opentitan/+/earlgrey_1.0.0:sw/host/provisioning/orchestrator/src/device_id.py;l=111) to complete the full Device ID that gets passed to the FT stage.

Reply:
Done. PTAL.